src/pages/academy/grading/subcomponents/GradingSubmissionsTable.tsx

```diff
@@ -128,15 +128,22 @@ const GradingSubmissionTable: React.FC<GradingSubmissionTableProps> = ({
     ...tableFilters.columnFilters
   ]);
 
-  // Polish: debounce this search, or have a onClick event listener search instead.
-  // Polish: if search value does not change content of submissions, do not reset page.
-  // not as easy as i thought it was with setTimeout.
-  const [searchValue, setSearchValue] = useState('');
+  const [visibleSearchValue, setVisibleSearchValue] = useState('');
+  const [debouncedSearchValue, setDebouncedSearchValue] = useState('');
+
+  // timeout delay could be a placed within a config.
+  useEffect(() => {
+    const changeSearch = setTimeout(() => {
+      setDebouncedSearchValue(visibleSearchValue);
+    }, 300);
+
+    return () => clearTimeout(changeSearch);
+  }, [visibleSearchValue]);
```

Comment:
This is a solution which I came up with using Blueprint's InputGroup, We could also implement a submit button alongside it to improve UI/UX experience 
<InputGroup
  style={{ borderRadius: '6px', width: 'clamp(200px, 20vw, 400px)' }}
  leftIcon={<BpIcon icon={IconNames.SEARCH} style={{ marginLeft: '0.75rem' }} />}
  placeholder="Search for assessment..."
  onKeyUp={e => {
    if (e.key === 'Enter') setSearchValue((e.target as HTMLInputElement).value);
  }}
/>